### PR TITLE
The light, IT BURNS!

### PR DESCRIPTION
**Pull request by monster860**
- Shadowlings get a fancy overbright effect when their darksight is enabled.
- Lighting overlays are put on two planes: PLANE_LIGHTING_G for greyscale overlays, PLANE_LIGHTING_C for colored overlays
- HUD objects get put on plane 20 or higher.
- Lighting overlays are changed to make it possible to do these effects. Clients now get a default plane master to get lighting to look correct.
- It's possible to make other PLANE_MASTER lighting effects in future PR's.
- In a future PR, we could abandon the use of invisibility vars for lighting overlays.
- I'm not copypasting /datum/camerachunk this time!

:cl: monster860
rscadd: Shadowlings get an overbright effect
/:cl:

### CODE REVIEW

**Comment by Regen1** [comment]
I like it :+1:

**Comment by FlattestGuitar** [comment]
Fancy. :+1: :bulb:

**Comment by tigercat2000** [comment]
:-1: :x: :100:
 - Adds plane definitions everywhere (it shouldn't be used unless actually necessary)
 - Snowflaked into multiple places with it's fullscreen procs
 - Looks horrible IMO
 - Changes the lighting system to set itself to pure white instead of black when "dark"

**Comment by fludd12** [comment]
I'm a fan of it.

**Comment by monster860** [comment]
Is there a better way?

**Comment by Aurorablade** [comment]
Soo besides blinding tea player Thier computer I do not a prActical or aesthic use for this

**Comment by Aurorablade** [comment]
Damn phone typeing

**Comment by Spacemanspark** [comment]
I personally like the concept.

**Comment by QQCucumber** [comment]
This looks awesome.

**Comment by NTSAM** [comment]
OH GOD IT BURNS
👍

**Comment by taukausanake** [comment]
I have never seen something so horrifying before. I like it a lot. Kind of wonder what Tiger means however

File: code/_onclick/hud/hud.dm [hunk 1]
```diff
@@ -155,6 +155,10 @@
 				mymob.client.screen -= hotkeybuttons
 			if(infodisplay.len)
 				mymob.client.screen -= infodisplay
+	
+	// Yes, snowflekkey, but I'd rather not hunt through every non-human mob hud.
+	mymob.overlay_fullscreen("pmaster_lighting_g", /obj/screen/fullscreen/pmaster_g_default, 1)
```
Comment (tigercat2000) [inline code/_onclick/hud/hud.dm]: put this in hud/New()

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 2]
```diff
@@ -461,7 +461,13 @@
 		H.see_in_dark = 8
 		if(!H.druggy)
 			H.see_invisible = SEE_INVISIBLE_LEVEL_TWO
+		if(H.client)
+			H.overlay_fullscreen("pmaster_lighting_g", /obj/screen/fullscreen/pmaster_g_default, 1)
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/species/species.dm]: no

Comment (monster860) [inline code/modules/mob/living/carbon/human/species/species.dm]: It really doesn't help when you just say "no" and nothing else. I'm not sure exactly what you are expressing disapproval to. Is it the way I called this function, is it where it is, or what?

Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/species/species.dm]: you call this exact proc three times, when overlay_fullscreen should only be called when shit actually needs to be updated

**Comment by monster860** [comment]
@TheDZD Morgue conflicts fixed.

**Comment by Fox-McCloud** [comment]
For the reasons stated above, namely:
- Snowflaked into several places
- Planes defined into a whole bunch of areas
- White shadows

This is going to be closed

Additionally, this really isn't necessary since Shadowlings now have toggle-able nightvision; they can see full bright or easily see where the light actually is.

**Pull request closed without merging**